cleanlab/datalab/internal/issue_manager_factory.py

```diff
@@ -143,14 +169,30 @@ def find_issues(self, **kwargs):
 
         register(MyIssueManager)
     """
-    name: str = str(cls.issue_name)
-    if name in REGISTRY:
-        # Warn user that they are overwriting an existing issue manager
-        print(
-            f"Warning: Overwriting existing issue manager {name} with {cls}. "
-            "This may cause unexpected behavior."
-        )
+
     if not issubclass(cls, IssueManager):
         raise ValueError(f"Class {cls} must be a subclass of IssueManager")
+
+    name: str = str(cls.issue_name)
+
+    if task is None:
+        if name in REGISTRY:
+            print(
+                f"Warning: Overwriting existing issue manager {name} with {cls}. "
+                "This may cause unexpected behavior."
+            )
+
+    else:
+        if task not in TASK_SPECIFIC_REGISTRY:
+            raise ValueError(
+                f"Invalid task type: {task}, must be in {list(TASK_SPECIFIC_REGISTRY.keys())}"
+            )
```

Comment:
Consider moving this to the top to raise an error early in the function.

Reply:
Sure, will be covered be nesting refactoring.